sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -104,6 +117,123 @@ class SentryOkHttpInterceptor(
+
+//            // with description, mechanism in the UI is buggy
+//            description = message
+
+            // TODO: should it be synthetic? likely not, mechanism SentryOkHttpInterceptor should be considered for grouping
+//            synthetic = true
+        }
+        val exception = SentryHttpClientError("Event was captured because the request status code was ${response.code}")
+        // TODO: okhttp thread uses thread pool so its always the same stack trace (okhttp frames only), stacktrace is set as snapshot=true, is that ok?
+        val mechanismException = ExceptionMechanismException(mechanism, exception, Thread.currentThread(), true)
+        val event = SentryEvent(mechanismException)
+
+        // TODO: Do we need a message if the request and response already have the info?
+        val sentryMessage = Message().apply {
+            message = "HTTP url: %s status: %s"
+            params = listOf(requestUrl, response.code.toString())
+        }
+
+        val hint = Hint()
+        hint.set("request", request)
+        hint.set("response", response)
+
+        // TODO: remove after fields indexed
+        val tags = mutableMapOf<String, String>()
+        tags["status_code"] = response.code.toString()
+        tags["url"] = requestUrl
+
+        val unknownRequestFields = mutableMapOf<String, Any>()
+
+        val sentryRequest = io.sentry.protocol.Request().apply {
+            url = requestUrl
+            cookies = request.headers["Cookie"]
+            method = request.method
+            queryString = query
+            headers = getHeaders(request.headers)
+            fragment = urlFragment
+
+            request.body?.contentLength().ifHasValidLength {
+                // TODO: should be mapped in relay and added to the protocol
+                unknownRequestFields["body_size"] = it
+            }
+
+            unknown = unknownRequestFields.ifEmpty { null }
+        }
+
+        val sentryResponse = io.sentry.protocol.Response().apply {
+            cookies = response.headers["Cookie"]
+            headers = getHeaders(response.headers)
+            statusCode = response.code
+
+            response.body?.contentLength().ifHasValidLength {
+                bodySize = it
+            }
+        }
+
+        event.tags = tags
+        event.request = sentryRequest
+        event.contexts.setResponse(sentryResponse)
+        event.message = sentryMessage
+
+        hub.captureEvent(event, hint)
+    }
+
+    private fun containsStatusCode(statusCode: Int): Boolean {
+        for (item in failedRequestStatusCode) {
+            if (item.isInRange(statusCode)) {
+                return true
+            }
+        }
+        return false
+    }
+
+    private fun getHeaders(requestHeaders: Headers): MutableMap<String, String>? {
+        // TODO: should it be under isSendDefaultPii?
+        // Server already does data scrubbing
+        if (!hub.options.isSendDefaultPii) {
+            return null
+        }
```

Comment:
@jjbayer I believe its fine to send them by default since the server is going to data scrub the headers, right?

Reply:
As discussed offline, if that is what most SDKs do for request headers, then yes, let's not check `isSendDefaultPii` for either request headers or response headers, and let relay scrub them.

Reply:
Sorry I've seen the wrong class for Spring
https://github.com/getsentry/sentry-java/blob/4ca1d7b2278581c9c73fa0558610dbac45a52682/sentry-spring/src/main/java/io/sentry/spring/SentryRequestResolver.java#L49-L52
So it does only send if `isSendDefaultPii` is enabled, so I'd rather mimic the behavior for the sake of compatibility, we can decide to remove this guard separately, for every integration.